control/controlclient/map.go

```diff
@@ -198,32 +199,30 @@ func (ms *mapSession) HandleNonKeepAliveMapResponse(ctx context.Context, resp *t
 	return nil
 }
 
-func (ms *mapSession) addUserProfile(userID tailcfg.UserID) {
-	if userID == 0 {
-		return
-	}
-	nm := ms.netMapBuilding
-	if _, dup := nm.UserProfiles[userID]; dup {
-		// Already populated it from a previous peer.
-		return
-	}
-	if up, ok := ms.lastUserProfile[userID]; ok {
-		nm.UserProfiles[userID] = up
-	}
+// updateStats are some stats from updateStateFromResponse, primarily for
+// testing. It's meant to be cheap enough to always compute, though. It doesn't
+// allocate.
+type updateStats struct {
+	allNew  bool
+	added   int
+	removed int
+	changed int
 }
 
-// netmapForResponse returns a fully populated NetworkMap from a full
-// or incremental MapResponse within the session, filling in omitted
-// information from prior MapResponse values.
-func (ms *mapSession) netmapForResponse(resp *tailcfg.MapResponse) *netmap.NetworkMap {
-	undeltaPeers(resp, ms.previousPeers)
+// updateStateFromResponse updates ms from res. It takes ownership of res.
+func (ms *mapSession) updateStateFromResponse(res *tailcfg.MapResponse) {
```

Comment:
rename is a bit distracting

Reply:
Well, it's more than a rename. The func was broke in two halves, necessarily for a future change.

Reply:
i mean the `resp` -> `res` rename

Reply:
One half updates internal state only. (in the future some lite updates will only do that step, and not generate a big netmap).

The second half allocs the NetworkMap when needed. (currently always)

Reply:
Oh, I can put `resp` back.